server/controllers/board.js

```diff
@@ -20,49 +22,64 @@ exports.listOneBoard = asyncHandler(async (req, res, next) => {
     res.status(404);
     throw new Error("No board found with given id");
   }
-  const lastViewedBoard = await Board.findOne({user: req.user.id, lastViewed: true})
-  if (lastViewedBoard && board._id != lastViewedBoard._id) {
-    lastViewedBoard.lastViewed = false
-    await lastViewedBoard.save();
+
+  /**
+   * if current board is not lastViewed, first we try to set to false any
+   * board with lastViewed, Second set the current board to lastViewed.
+   *
+   * OBS: if current board is already lastViewed. just pass through.
+   */
+  if (board.lastViewed === false) {
+    await Board.findOneAndUpdate({user: req.user.id, lastViewed: true}, {
+      lastViewed: false,
+    })
+
+    await Board.findOneAndUpdate({user: req.user.id, _id: board._id}, {
+      lastViewed: true,
+    })
   }
```

Comment:
Perhaps we could instead of setting this property on the Board but rather on the User, when they login we can see the id of the board they last viewed and show that.